merge.md

```diff
@@ -0,0 +1,136 @@
+## Merge and override
+
+Compose implementations SHOULD allow users to define a Compose application model through multiple compose files. 
+When doing so, a Compose implementation MUST follow the rules declared in this section to merge Compose files.
+
+### Mapping
+
+A YAML `mapping` gets merged by adding missing entries and merging the conflicting ones.
+
+Merging the following example YAML trees:
+```yaml
+services:
+  foo:
+    key1: value1
+    key2: value2
+```    
+
+```yaml
+services:
+  foo:
+    key2: VALUE
+    key3: value3
+```
+
+MUST result in a Compose application model equivalent to the YAML tree:
+
+```yaml
+services:
+  foo:
+    key1: value1
+    key2: VALUE
+    key3: value3
+```
+
+### Sequence
+
+A YAML `sequence` is merged by appending values from the overriding Compose file to the previous one.
+
+Merging the following example YAML trees:
+```yaml
+services:
+  foo:
+    DNS:
+      - 1.1.1.1
+```    
+
+```yaml
+services:
+  foo:
+    DNS: 
+      - 8.8.8.8
+```
+
+MUST result in a Compose application model equivalent to the YAML tree:
+
+```yaml
+services:
+  foo:
+    DNS:
+      - 1.1.1.1
+      - 8.8.8.8
+```
+
+## Exceptions
+
+There are exceptions to those rules:
+
+### Shell commands
+
+Service's [command](#command), [entrypoint](#entrypoint) and [healthcheck](#healthcheck) `test`: 
+For usability, the value MUST be overridden by the latest Compose file, and not appended.
+
+Merging the following example YAML trees:
+```yaml
+services:
+  foo:
+    command: ["echo", "foo"]
+```    
+
+```yaml
+services:
+  foo:
+    command: ["echo", "bar"]
+```
+
+MUST result in a Compose application model equivalent to the YAML tree:
+
+```yaml
+services:
+  foo:
+    DNS:
+      command: ["echo", "bar"]
+```
+
+
+### Unique resources
+
+Service's [ports](#ports), [volumes](#volumes), [secrets](#secrets) and [configs](#configs). While those
+are modeled in a Compose file as a sequence, they MUST be unique regarding the target resource in container:
+
+| attribute   | unique key               |
+|-------------|--------------------------|
+| volumes     |  target                  |
+| secrets     |  source                  |
+| configs     |  source                  |
+| ports       |  {ip, target, published, protocol}   |
+
+While merging Compose files, a Compose implementation MUST respect these unique keys and merge the entries
+in sequence just like this would have been a mapping.
```

Comment:
This is a bit hard to interpret. I tried to write an alternative but not claiming it's easy to understand either 🤪 

```suggestion
While merging Compose files, a Compose implementation MUST append new entries that do not violate a uniqueness constraint and merge entries that share a unique key.
```

Reply:
Yes, I also got issue to explain this. I wonder we should introduce support for a mapping-style syntax on ports, volumes, secrets, config so that this "unicity" concept is more obvious. I will prepare a PR for this purpose